experiment/compatibility-versions/kind-upgrade.sh

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# script adapted from - https://gist.github.com/aojea/2c94034f8e86d08842e5916231eb3fe1
+
+set -e
+set -o pipefail
+
+build_docker(){
+  build/run.sh make all WHAT="cmd/kubectl cmd/kubelet" 1> /dev/null
+  make quick-release-images 1> /dev/null
+}
+
+update_kubelet() {
+ for n in $NODES; do
+   # Backup previous kubelet
+   docker exec $n cp /usr/bin/kubelet /usr/bin/kubelet.bak
+   # Install new kubelet binary
+   docker cp ${KUBELET_BINARY} $n:/usr/bin/kubelet
+   docker exec $n systemctl restart kubelet
+  echo "Updated kubelet on node $n"
+ done
+}
+
+update_kube_proxy() {
+  for n in $NODES; do
+   kind load image-archive ${IMAGES_PATH}/kube-proxy.tar --name ${CLUSTER_NAME}
+  done
+  # RollingUpdate
+  kubectl set image ds/kube-proxy kube-proxy=${DOCKER_REGISTRY}/kube-proxy-amd64:${DOCKER_TAG} -n kube-system
+  kubectl rollout status ds/kube-proxy -n kube-system -w
+  echo "Updated kube-proxy"
+}
+
+update_cni() {
```

Comment:
do we need all of these? or just kubelet? I think most of these won't be used?

Reply:
Removed kube-proxy and cni from the kind-upgrade.sh script for now.  Technically we could remove all args from the script and remove the kubelet functionality as well (as we turn it off) but leaving it for now as it makes what the test does more explicit (doesn't upgrade kubelet).